mongo-c-driver installation failure on Ubuntu 18.04 x64

Freshly installed and build vcpkg from the repo as of the time of this writing.

```
./vcpkg install mongo-c-driver
The following packages will be built and installed:
  * libbson[core]:x64-linux
    mongo-c-driver[core]:x64-linux
Additional packages (*) will be modified to complete this operation.
Starting package 1/2: libbson:x64-linux
Building package libbson[core]:x64-linux... 
-- Downloading https://github.com/mongodb/libbson/archive/1.9.2.tar.gz...
-- Extracting source /home/alex/vcpkg/downloads/libbson-1.9.2.tar.gz
-- Applying patch /home/alex/vcpkg/ports/libbson/fix-uwp.patch
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
CMake Error at ports/libbson/portfile.cmake:48 (file):
  file RENAME failed to rename

    /home/alex/vcpkg/packages/libbson_x64-linux/lib/bson-static-1.0.lib

  to

    /home/alex/vcpkg/packages/libbson_x64-linux/lib/bson-1.0.lib

  because: No such file or directory

Call Stack (most recent call first):
  scripts/ports.cmake:71 (include)


Error: Building package libbson:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libbson:x64-linux
  Vcpkg version: 0.0.113-unknownhash
```

I tried to install it with latest vcpkg version, and it failed with below issue:

linux001@linuxtest01:~/vcpkg/buildtrees/mongo-c-driver$ cat ./config-x64-linux-dbg-err.log
CMake Warning at CMakeLists.txt:9 (_message):
  CMake support is experimental and may not produce production quality
  artifacts
Call Stack (most recent call first):
  CMakeLists.txt:37 (message)
Current version (from VERSION_CURRENT file): 1.9.5
--   libbson-static found version "1.9.5"
--   libbson-static include path "/home/linux001/vcpkg/installed/x64-linux/include"
--   libbson-static libraries "/home/linux001/vcpkg/installed/x64-linux/lib/libbson-1.0.a-lgcc;-lgcc_s;-lc;-lgcc;-lgcc_s;/usr/lib/x86_64-linux-gnu/librt.so"
CMake Error at CMakeLists.txt:9 (_message):
  ENABLE_SSL=WINDOWS only supported on Windows
Call Stack (most recent call first):
  CMakeLists.txt:134 (message)

Other information:
Starting package 2/2: mongo-c-driver:x64-linux
Building package mongo-c-driver[core]:x64-linux...
-- Downloading https://github.com/mongodb/mongo-c-driver/archive/1.9.5.tar.gz...
-- Extracting source /home/linux001/vcpkg/downloads/mongodb-mongo-c-driver-1.9.5.tar.gz
-- Applying patch fix-uwp.patch
-- Using source at /home/linux001/vcpkg/buildtrees/mongo-c-driver/src/1.9.5-26879286b7
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: /home/linux001/vcpkg/downloads/tools/cmake-3.12.4-linux/cmake-3.12.4-Linux-x86_64/bin/cmake;/home/linux001/vcpkg/buildtrees/mongo-c-driver/src/1.9.5-26879286b7;-DBSON_ROOT_DIR=/home/linux001/vcpkg/installed/x64-linux;-DENABLE_TESTS=OFF;-DENABLE_EXAMPLES=OFF;-DENABLE_SSL=WINDOWS;-DENABLE_STATIC=ON;-DCMAKE_MAKE_PROGRAM=/home/linux001/vcpkg/downloads/tools/ninja-1.8.2-linux/ninja;-DCMAKE_SYSTEM_NAME=Linux;-DBUILD_SHARED_LIBS=OFF;-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/linux001/vcpkg/scripts/toolchains/linux.cmake;-DVCPKG_TARGET_TRIPLET=x64-linux;-DVCPKG_PLATFORM_TOOLSET=external;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON;-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE;-DCMAKE_VERBOSE_MAKEFILE=ON;-DVCPKG_APPLOCAL_DEPS=OFF;-DCMAKE_TOOLCHAIN_FILE=/home/linux001/vcpkg/scripts/buildsystems/vcpkg.cmake;-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON;-DVCPKG_CXX_FLAGS=;-DVCPKG_CXX_FLAGS_RELEASE=;-DVCPKG_CXX_FLAGS_DEBUG=;-DVCPKG_C_FLAGS=;-DVCPKG_C_FLAGS_RELEASE=;-DVCPKG_C_FLAGS_DEBUG=;-DVCPKG_CRT_LINKAGE=dynamic;-DVCPKG_LINKER_FLAGS=;-DCMAKE_INSTALL_LIBDIR:STRING=lib;-DCMAKE_INSTALL_BINDIR:STRING=bin;-G;Ninja;-DCMAKE_BUILD_TYPE=Debug;-DCMAKE_INSTALL_PREFIX=/home/linux001/vcpkg/packages/mongo-c-driver_x64-linux/debug
    Working Directory: /home/linux001/vcpkg/buildtrees/mongo-c-driver/x64-linux-dbg
    See logs for more information:
      /home/linux001/vcpkg/buildtrees/mongo-c-driver/config-x64-linux-dbg-out.log
      /home/linux001/vcpkg/buildtrees/mongo-c-driver/config-x64-linux-dbg-err.log
Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:265 (vcpkg_execute_required_process)
  ports/mongo-c-driver/portfile.cmake:22 (vcpkg_configure_cmake)
  scripts/ports.cmake:71 (include)
Error: Building package mongo-c-driver:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mongo-c-driver:x64-linux
  Vcpkg version: 2018.11.23-unknownhash

--- a/ports/mongo-c-driver/CONTROL
+++ b/ports/mongo-c-driver/CONTROL
@@ -1,4 +1,4 @@
 Source: mongo-c-driver
-Version: 1.9.5-1
+Version: 1.9.5-2
 Build-Depends: libbson, openssl (uwp)
 Description: Client library written in C for MongoDB.

--- a/ports/mongo-c-driver/fix-uwp.patch
+++ b/ports/mongo-c-driver/fix-uwp.patch
@@ -90,19 +90,26 @@ index 4e181bd..b274fc1 100644
     ZeroMemory (&osvi, sizeof (OSVERSIONINFO));
     osvi.dwOSVersionInfoSize = sizeof (OSVERSIONINFO);
 diff --git a/src/mongoc/mongoc-util.c b/src/mongoc/mongoc-util.c
-index 6dbb26d..7f33780 100644
+index a4e50ea..4f631e7 100644
 --- a/src/mongoc/mongoc-util.c
 +++ b/src/mongoc/mongoc-util.c
-@@ -69,7 +69,11 @@ _mongoc_hex_md5 (const char *input)
- void
+@@ -70,6 +70,11 @@ void
  _mongoc_usleep (int64_t usec)
  {
--#ifdef _WIN32
+ #ifdef _WIN32
 +#if defined (WINAPI_FAMILY_PARTITION) && !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 +   if (usec / 1000 > MAXDWORD || usec < 0)
 +      __fastfail(-1);
 +   Sleep((DWORD)(usec / 1000));
-+#elif defined(_WIN32)
++#else
     LARGE_INTEGER ft;
     HANDLE timer;
- 
+
+@@ -80,6 +85,7 @@ _mongoc_usleep (int64_t usec)
+    SetWaitableTimer (timer, &ft, 0, NULL, NULL, 0);
+    WaitForSingleObject (timer, INFINITE);
+    CloseHandle (timer);
++#endif
+ #else
+    BSON_ASSERT (usec >= 0);
+    usleep ((useconds_t) usec);

--- a/ports/mongo-c-driver/portfile.cmake
+++ b/ports/mongo-c-driver/portfile.cmake
@@ -14,8 +14,9 @@ else()
     set(ENABLE_STATIC OFF)
 endif()
 
-set(ENABLE_SSL "WINDOWS")
-if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+if(NOT VCPKG_CMAKE_SYSTEM_NAME)
+    set(ENABLE_SSL "WINDOWS")
+else()
     set(ENABLE_SSL "OPENSSL")
 endif()
 
@@ -48,12 +49,21 @@ file(RENAME ${CURRENT_PACKAGES_DIR}/temp ${CURRENT_PACKAGES_DIR}/include)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
-    file(RENAME
-        ${CURRENT_PACKAGES_DIR}/lib/mongoc-static-1.0.lib
-        ${CURRENT_PACKAGES_DIR}/lib/mongoc-1.0.lib)
-    file(RENAME
-        ${CURRENT_PACKAGES_DIR}/debug/lib/mongoc-static-1.0.lib
-        ${CURRENT_PACKAGES_DIR}/debug/lib/mongoc-1.0.lib)
+    if(VCPKG_CMAKE_SYSTEM_NAME AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+	    file(RENAME
+            ${CURRENT_PACKAGES_DIR}/lib/libmongoc-static-1.0.a
+            ${CURRENT_PACKAGES_DIR}/lib/libmongoc-1.0.a)
+        file(RENAME
+            ${CURRENT_PACKAGES_DIR}/debug/lib/libmongoc-static-1.0.a
+            ${CURRENT_PACKAGES_DIR}/debug/lib/libmongoc-1.0.a)
+    else()
+        file(RENAME
+            ${CURRENT_PACKAGES_DIR}/lib/mongoc-static-1.0.lib
+            ${CURRENT_PACKAGES_DIR}/lib/mongoc-1.0.lib)
+        file(RENAME
+            ${CURRENT_PACKAGES_DIR}/debug/lib/mongoc-static-1.0.lib
+            ${CURRENT_PACKAGES_DIR}/debug/lib/mongoc-1.0.lib)
+    endif()
 
     # drop the __declspec(dllimport) when building static
     vcpkg_apply_patches(

